tests/tasks/setup/kitctl/setup-control-plane.yaml

```diff
@@ -0,0 +1,124 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: control-plane-setup
+  namespace: tekton-pipelines
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/categories: Kubernetes
+    tekton.dev/tags: CLI, kubectl
+    tekton.dev/displayName: "kubernetes actions"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: |
+    Setup a kubernetes control plane in the guess cluster.
+  params:
+  - name: name
+    default: "guest"
+    description: Name of the guess cluster
+  - name: scheduler-replicas
+    default: "1"
+    description: Number of scheduler replicas
+  - name: controller-manager-replicas
+    default: "1"
+    description: Number of controller manager replicas
+  - name: apiserver-replicas
+    default: "1"
+    description: Number of APIserver replicas
+  - name: apiserver-image
+    default: "public.ecr.aws/eks-distro/kubernetes/kube-apiserver:v1.21.9-eks-1-21-11"
+    description: Image of apiserver
+  - name: apiserver-parameters
+    default: "[]"
+    description: Parameters of the apiserver container
+  - name: apiserver-instance-type
+    default: "m5.16xlarge"
+    description: Instance type for the apiserver
+  - name: etcd-replicas
+    default: "3"
+    description: Number of ETCD replicas
+  - name: etcd-image
+    default: "public.ecr.aws/eks-distro/etcd-io/etcd:v3.4.18-eks-1-21-11"
+    description: Image of ETCD
+  - name: etcd-parameters
+    default: "[]"
+    description: Parameters of the ETCD container
+  - name: etcd-instance-type
+    default: "m5.16xlarge"
+    description: Instance type for the ETCD
+  - name: kubernetes-version
+    default: "1.19"
+    description: Kubernetes version for the guest cluster 
+  steps:
+  - name: setup-control-plane
+    image: gcr.io/cloud-builders/kubectl@sha256:8ab94be8b2b4f3d117f02d868b39540fddd225447abf4014f7ba4765cb39f753
+    script: |
+      #!/bin/bash
+      echo "Approving KCM requests"
+      kubectl certificate approve $(kubectl get csr | grep "Pending" | awk '{print $1}')  2>/dev/null || true
```

Comment:
Do we need it here, we are already doing it in the end?

Reply:
This will unblock Tekton dashboard from getting status update if the tekton node is dynamically provisioned by Karpenter.